tests/benchdnn/graph/bench_graph.cpp

```diff
@@ -65,13 +67,23 @@ int bench(int argc, char **argv) {
     for (; argc > 0; --argc, ++argv) {
         const bool parsed_options = parse_bench_settings(argv[0])
                 || parse_batch(bench, argv[0])
-                || parse_dt(s.dt, def.dt, argv[0])
+                || parse_dt(s.dt, s.dt_map, argv[0])
                 || parse_input_shapes(s.in_shapes_vec, argv[0])
                 || parse_op_attrs(s.op_attrs_vec, argv[0])
                 || parse_graph_expected_n_partitions(
                         s.expected_n_partition_vec, argv[0])
                 || parse_graph_fpmath_mode(s.fpmath_mode_vec, argv[0])
                 || parse_mb(s.mb, def.mb, argv[0]) || parse_reset(s, argv[0]);
+        // Graph driver does not support two styles of --dt in a single command line.
+        const bool has_dt = s.dt.size() != 1 || s.dt[0] != dnnl_data_type_undef;
+        const bool has_dt_map
+                = s.dt_map.size() != 1 || (s.dt_map[0].count(SIZE_MAX) == 0);
+        if (has_dt && has_dt_map) {
+            BENCHDNN_PRINT(0, "%s\n",
+                    "Error: --dt is specified twice with different styles.");
+            SAFE_V(FAIL);
+        }
```

Comment:
I would move it into `parse_dt` since it has both objects and can check their state to figure out if both options were passed. Afterall, it's a parser obligation to make sure the input is valid (when it can).

Reply:
Sure, it's moved to the parser function now.